src/wp-includes/media.php

```diff
@@ -5661,6 +5661,27 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 			}
 	}
 
+	/**
+	 * Filters whether to short-circuit loading optimization attributes.
+	 *
+	 * Returning an array from the filter will effectively short-circuit the loading of optimization attributes,
+	 * returning that value instead.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param false|array $loading_attrs The loading optimization attributes.
+	 * @param array       $attr          Array of the attributes for the tag.
+	 * @param string      $context       Context for the element for which the loading optimization attribute is requested.
+	 */
+	$loading_attrs = apply_filters( 'wp_pre_get_loading_optimization_attributes', false, $attr, $context );
+
+	if ( is_array( $loading_attrs ) ) {
+		return $loading_attrs;
+	}
+
+	// Resets the loading attributes.
+	$loading_attrs = array();
```

Comment:
normal filters like this are at the state of the function. There is mulitple return statements at the state of this function, meaning this would not fire.

Reply:
I agree this filter should be moved to the very top of the function, since that would bring several benefits such as:
* A plugin could support more tags than `img` and `iframe`.
* A plugin could add optimization logic for the `template` context (which is for the overall block template).
* A plugin could add optimization logic that is not tied to `width` and `height` attributes (which in fact already applies to `decoding="async"` (see #4991).